persist/sqldb/backfill_cluster_name.go

```diff
@@ -0,0 +1,56 @@
+package sqldb
+
+import (
+	"fmt"
+
+	log "github.com/sirupsen/logrus"
+	"upper.io/db.v3"
+	"upper.io/db.v3/lib/sqlbuilder"
+)
+
+type backfillClusterName struct {
+	clusterName string
+	tableName   string
+}
+
+func (s backfillClusterName) String() string {
+	return fmt.Sprintf("backfillClusterName{%s,%s}", s.clusterName, s.tableName)
+}
+
+func (s backfillClusterName) Apply(session sqlbuilder.Database) error {
+	log.WithField("clustername", s.clusterName).Info("Back-filling cluster name")
+	rs, err := session.
+		Select("uid").
+		From(s.tableName).
+		Where(db.Cond{"clustername": nil}).
+		Query()
+	if err != nil {
+		return err
+	}
+	for rs.Next() {
+		uid := ""
+		err := rs.Scan(&uid)
+		if err != nil {
+			return err
+		}
+		logCtx := log.WithFields(log.Fields{"clustername": s.clusterName, "uid": uid})
+		logCtx.Info("Back-filling cluster name")
+		res, err := session.
+			Update(s.tableName).
+			Set("clustername", s.clusterName).
+			Where(db.Cond{"clustername": nil}).
+			And(db.Cond{"uuid": uid}).
```

Comment:
should this line read the following?
```
And(db.Cond{"uid": uid}).
```

Reply:
yes - though why exactly any SQL framework would not throw some kind of exception at this clear error is beyond me 😠 😠 😠 😠

Reply:
actually - maybe this code never runs

Reply:
It's running for me on migrating a legacy database ...

![image](https://user-images.githubusercontent.com/192430/73674578-bdbdbd00-46a8-11ea-8d2b-5334c89f2b8e.png)

Reply:
did it migrate successfully? give that it'l likely that every record should have "default" it may be just luckily - though I ask again - who writes framework that allows such a big error thought?

Reply:
no reviewing the code - this backfill is really implemented stupidly - I think I'll make a patch fix and then a re-write

Reply:
https://github.com/argoproj/argo/pull/2141